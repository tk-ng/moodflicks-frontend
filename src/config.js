const PosterBaseUrl = "https://image.tmdb.org/t/p/";

export { PosterBaseUrl };
