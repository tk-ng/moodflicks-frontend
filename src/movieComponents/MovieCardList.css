.MovieCard {
	background-color: var(--card-bg);
	color: var(--text-color);
	border: 20px solid rgba(16, 16, 16, 0.2);
	box-shadow: inset 0 0 0 2px var(--shadow-color);
}
.MovieCard:hover {
	cursor: pointer;
}

.MovieCardTruncate {
	display: -webkit-box;
	-webkit-line-clamp: 3;
	-webkit-box-orient: vertical;
	overflow: hidden;
	text-overflow: ellipsis;
	text-align: start;
}
