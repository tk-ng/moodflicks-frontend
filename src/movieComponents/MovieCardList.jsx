import { MovieCard } from "./MovieCard";
import "./MovieCardList.css";

export const MovieCardList = ({ movies, editAuth = false, handleRemove }) => {
	return (
		<div className="MovieCardList d-flex flex-wrap justify-content-evenly">
			{movies.map((m) => (
				<MovieCard
					posterPath={m.posterPath}
					id={m.id}
					key={m.id}
					overview={m.overview}
					title={m.title}
					releaseDate={m.releaseDate}
					editAuth={editAuth}
					handleRemove={handleRemove}
				/>
			))}
		</div>
	);
};
