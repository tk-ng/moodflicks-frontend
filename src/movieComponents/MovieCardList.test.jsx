import { MovieCardList } from "./MovieCardList";
import { MemoryRouter } from "react-router";
import { render, screen } from "@testing-library/react";
import { PosterBaseUrl } from "../config";
import userEvent from "@testing-library/user-event";

describe("Render the MovieCardList component", function () {
	const handleRemove = vi.fn();
	const testMovies = [
		{
			id: "testMovieId1",
			overview: "testMovieOverview1",
			posterPath: "testMoviePosterPath1",
			releaseDate: "testMovieReleaseDate1",
			title: "testMovieTitle1",
		},
		{
			id: "testMovieId",
			overview: "testMovieOverview",
			posterPath: "testMoviePosterPath",
			releaseDate: "testMovieReleaseDate",
			title: "testMovieTitle",
		},
	];
	it("mounts without crashing", async function () {
		const { container } = render(
			<MemoryRouter>
				<MovieCardList movies={testMovies} editAuth={false} />
			</MemoryRouter>
		);
		expect(container).toBeInTheDocument();
	});

	it("matches snapshot", async function () {
		const { asFragment } = render(
			<MemoryRouter>
				<MovieCardList movies={testMovies} editAuth={false} />
			</MemoryRouter>
		);
		expect(asFragment).toMatchSnapshot();
	});

	it("renders movie details WITHOUT the Remove buttons", async function () {
		render(
			<MemoryRouter>
				<MovieCardList movies={testMovies} editAuth={false} />
			</MemoryRouter>
		);
		expect(
			screen.getByText(testMovies[0].title, { selector: "h5" })
		).toBeInTheDocument();

		expect(
			screen.getByText(`released: ${testMovies[0].releaseDate}`, {
				selector: "h6",
			})
		).toBeInTheDocument();

		// The Remove button should not be displayed
		expect(
			screen.queryByText("Remove", { selector: "button" })
		).not.toBeInTheDocument();

		// Movie posters are displayed for each movie
		screen
			.getAllByAltText("Movie Poster")
			.forEach((el, idx) =>
				expect(el).toHaveAttribute(
					"src",
					PosterBaseUrl + "w185" + testMovies[idx].posterPath
				)
			);
	});
});
