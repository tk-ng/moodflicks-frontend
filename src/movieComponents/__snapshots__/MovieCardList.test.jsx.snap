// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render the MovieCardList component > matches snapshot 1`] = `[Function]`;
